src/docs/release/breaking-changes/test-widgets-flutter-binding-clock.md

```diff
@@ -0,0 +1,104 @@
+---
+title: `TestWidgetsFlutterBinding.clock` change
+description: The `Clock` implementation will now come from package:clock.
+---
+
+## Summary
+
+The `TestWidgetsFlutterBinding.clock` will now come from `package:clock` and
+not `package:quiver`.
+
+## Context
+
+The `flutter_test` package is removing its dependency on the heavier weight
+`quiver` package in favor of a dependency on two more targeted and lighter
+weight packages, `clock` and `fake_async`.
+
+This can affect user code which grabs the clock from a
+`TestWidgetsFlutterBinding` and passes that to an api that expects a `Clock`
+from `package:quiver`, for example some code like this:
+
+<!-- skip -->
+```dart
+testWidgets('some test', (WidgetTester tester) {
+  someApiThatWantsAQuiverClock(tester.binding.clock);
+});
+```
+
+## Migration guide
+
+### Option #1: Create a packge:quiver Clock from a package:clock Clock
+
+The easiest migration is to create a `package:quiver` clock from the
+`package:clock` clock, which can be done by passing the `.now` function
+tearoff to the `Clock` constructor:
+
+Code before migration:
+
+<!-- skip -->
+```dart
+testWidgets('some test', (WidgetTester tester) {
+  someApiThatWantsAQuiverClock(tester.binding.clock);
+});
+```
+
+Code after migration:
+
+<!-- skip -->
+```dart
+testWidgets('some test', (WidgetTester tester) {
+  someApiThatWantsAQuiverClock(Clock(tester.binding.clock.now));
+});
+```
+
+### Option #2: Change the api to accept a package:clock Clock
+
+If you own the api you are calling, you may want to change it to accept a Clock
+from `package:clock`. This is a judgement call based on how many places are
+calling this api with something other than a clock retreived from a
+`TestWidgetsFlutterBinding`.
+
+If you go this route your call sites that are passing `tester.binding.clock`
+will not need to be modified, but other call sites will.
+
+### Option #3: Change the api to accept a `DateTime function()`
+
+If all you use the `Clock` for is the `now` function, and you control the api,
+then you can also change it to accept that function directly instead of a
+`Clock`. This makes it easily callable with either type of `Clock`, by passing
+a tearoff of the `now` method from either type of clock:
+
+Calling code before migration:
+
+<!-- skip -->
+```dart
+testWidgets('some test', (WidgetTester tester) {
+  someApiThatWantsAQuiverClock(tester.binding.clock);
+});
+```
+
+Calling code after migration:
+
+<!-- skip -->
```

Comment:
can we avoid the skip on new code?

Reply:
I don't know what this is for - its in the template still though. It doesn't say why.

Reply:
it turns off the analysis of code snippets in CI

Reply:
(you want it for the "old" code, for obvious reasons, but if we can find a way to keep the "new" code analyzed that'd be great)

Reply:
Interesting, I assume `dart:ui` and flutter/flutter_test and all their transitive deps are in scope during analysis?

Reply:
Actually the point is that `quiver` will no longer be in the transitive deps, so I can't write analyzable code here (assuming only flutter and its transitive deps are available during analysis).